### PR TITLE
Update UI for Sign In by Syncing and Add Scan or Enter pairing Code to in app syncing flow

### PR DESCRIPTION
**Pull request by J-Son89**
The Onboarding Syncing flow and In app syncing flow is the same except for the background. 
This pr refactors the onboarding syncing screens so that it can be reused for both onboarding and in app syncing flows.

Also I adjust the onboarding syncing page as the background was not showing correctly.

To test:
Ensure the onboarding Syncing flows are working as expect. (i.e "sign in")

Also you can test the syncing flows by going 
profile -> Syncing -> big blue "+" button and then "Scan or enter pairing code" button at the bottom of the "Pair devices to sync" page.

Screenshot 2023-04-11 at 14 30 12

### CODE REVIEW

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        :width         (:width qr-view-finder)
+        :overflow      :hidden
+        :border-radius 16}
+       [camera-kit/camera
+        {:ref            #(reset! camera-ref %)
+         :style          style/camera-style
+         :camera-options {:zoomMode :off}
+         :scan-barcode   true
+         :on-read-code   on-read-code}]]
+      [rn/view {:width size :height size :justify-content :space-between}
```
Comment (jo-mut) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:106]: this alignment looks weird

**Review by jo-mut — APPROVED** [review]

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                                       :on-allowed  #(reset! camera-permission-granted? true)
+                                       :on-denied   #(rf/dispatch
+                                                      [:toasts/upsert
+                                                       {:icon :i/info
+                                                        :icon-color colors/danger-50
+                                                        :override-theme :light
+                                                        :text (i18n/label
+                                                               :t/camera-permission-denied)}])}]))]
+    [:f>
+     (fn []
```
Comment (flexsurfer) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs]: (fn [] should be never used in the hiccup. for proper usage, you have to rename `view` to `f-view` and used it with `[:f>`

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                 (let [layout      (js->clj (oops/oget event "nativeEvent.layout")
+                                            :keywordize-keys
+                                            true)
+                       view-finder (assoc layout :height (:width layout))]
+                   (reset! qr-view-finder view-finder)))}])
+
+(defn- border
+  [border1 border2 corner]
+  [rn/view
+   (assoc {:border-color colors/white :width 80 :height 80} border1 2 border2 2 corner 16)])
```
Comment (flexsurfer) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:98]: `{:style`

**Review by smohamedjavid — COMMENTED** [review]
@J-Son89 - The PR looks good!

I have one question regarding the behaviour of the Camera.

The camera renders within the viewfinder which is different to current UI behaviour of rendering the camera in fullscreen behind the blur. Is this a expected?

<table>
  <thead>
    <tr>
      <th>Existing behaviour</th>
      <th>New change</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td></td>
      <td></td>
    </tr>
  </tbody>
</table>

If we don't need to render the camera fullscreen, we can remove the `holeview` usage and `qr-view-finder` calculation.

**Comment by J-Son89** [comment]
> @J-Son89 - The PR looks good!
> 
> I have one question regarding the behaviour of the Camera.
> 
> The camera renders within the viewfinder which is different to current UI behaviour of rendering the camera in fullscreen behind the blur. Is this a expected?
> 
> Existing behaviour	New change
> 	
> If we don't need to render the camera fullscreen, we can remove the `holeview` usage and `qr-view-finder` calculation.

Thanks @smohamedjavid, you are right. However I think both of these versions are incorrect. 
In this pr's version when the camera is **not** enabled it looks correct (as far as the other onboarding screens look)

However in both current and this pr version when the camera is enabled it is a bit different in the designs. 
I think you are correct that it should render in full screen but it's the blur color which is not matching in the designs. 
I will try to get it to match closer to designs and if not I will sync with design team and give them their options :)

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+(defn render-background
+  [background show-camera? qr-view-finder]
+  (if (and show-camera? (:x qr-view-finder))
+    [hole-view/hole-view
+     {:style style/absolute-fill
+      :holes [(merge qr-view-finder
+                     {:borderRadius 16})]}
+     background]
+    background)
+)
```
Comment (ilmotta) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs]: The dreaded dangling parenthesis attacks again!

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+            (when-not @camera-permission-granted?
+              (permissions/permission-granted? :camera
+                                               #(reset! camera-permission-granted? %)
+                                               #(reset! camera-permission-granted? false)))))
+         [:<>
+          [render-background background show-camera? @qr-view-finder]
+          [rn/view {:style (style/root-container (:top insets))}
+           [header active-tab read-qr-once? title]
+           (case @active-tab
+             1 [scan-qr-code-tab qr-view-finder request-camera-permission
```
Comment (ilmotta) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs]: I think a bit better than `1` or `2` is to give meaningful names to the tabs, like `:tab/scan-qr-code` and `:tab/enter-sync-code`.

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+(defn- border
+  [border1 border2 corner]
+  [rn/view
+   (assoc {:border-color colors/white :width 80 :height 80} border1 2 border2 2 corner 16)])
+
+(defn- viewfinder
+  [qr-view-finder {:keys [camera-ref on-read-code]}]
+  (let [size (:width qr-view-finder)]
+    [:<>
```
Comment (ilmotta) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:103]: This fragment `:<>` seems unnecessary.

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 4 above)
Comment (J-Son89) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs]: 😅

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 5 above)
Comment (J-Son89) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs]: Yep that's a good idea 👍

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 6 above)
Comment (J-Son89) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:103]: I'll check it out

**Review by ilmotta — APPROVED** [review]

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:106]: should set styles in `:style` prop

**Review by smohamedjavid — APPROVED** [review]
Except the minor comments from other reviewers,
Looks good to me 🚀 .

**Comment by VolodLytvynenko** [comment]
hi @J-Son89 . Thank you for PR. No issues from my side. PR is ready to be merged

**Comment by J-Son89** [comment]
> hi @J-Son89 . Thank you for PR. No issues from my side. PR is ready to be merged

thanks @VladimrLitvinenko! :)

**Pull request merged**